code/modules/mafia/roles.dm

```diff
@@ -365,10 +443,80 @@
+		return
+	if(game_status == MAFIA_ALIVE && R && R.game_status == MAFIA_ALIVE)
+		if(!R.kill(source))
+			to_chat(body,"<span class='danger'>Your attempt at killing [R.body] was prevented!</span>")
+
+/datum/mafia_role/nightmare
+	name = "Nightmare"
+	desc = "You're a solo monster that cannot be detected by detective roles. You can flicker lights of another room each night. You can instead decide to hunt, killing everyone in a flickering room. Kill everyone to win."
+	win_condition = "kill everyone."
+	revealed_outfit = /datum/outfit/mafia/nightmare
+	detect_immune = TRUE
+	team = MAFIA_TEAM_SOLO
+	role_type = NEUTRAL_KILL
+	targeted_actions = list("Flicker", "Hunt")
+	var/list/flickering = list()
+	var/datum/mafia_role/flicker_target
+
+/datum/mafia_role/nightmare/New(datum/mafia_controller/game)
+	. = ..()
+	RegisterSignal(game,COMSIG_MAFIA_NIGHT_KILL_PHASE,.proc/flicker_or_hunt)
+
+/datum/mafia_role/nightmare/check_total_victory(alive_town, alive_mafia) //nightmares just want teams dead
+	return alive_town + alive_mafia <= 1
+
+/datum/mafia_role/nightmare/block_team_victory(alive_town, alive_mafia) //no team can win until they're dead
+	return TRUE //while alive, town AND mafia cannot win (though since mafia know who is who it's pretty easy to win from that point)
+
+/datum/mafia_role/nightmare/special_reveal_equip()
+	body.underwear = "Nude"
+	body.undershirt = "Nude"
+	body.socks = "Nude"
+	body.set_species(/datum/species/shadow)
+	body.update_body()
+
+/datum/mafia_role/nightmare/validate_action_target(datum/mafia_controller/game, action, datum/mafia_role/target)
+	. = ..()
+	if(!. || game.phase != MAFIA_PHASE_NIGHT || target.game_status != MAFIA_ALIVE)
+		return FALSE
+	if(action == "Flicker")
+		return target != src && !(target in flickering)
+	return target == src
+
+/datum/mafia_role/nightmare/handle_action(datum/mafia_controller/game, action, datum/mafia_role/target)
+	. = ..()
+	if(target == flicker_target)
+		to_chat(body,"<span class='warning'>You will do nothing tonight.</span>")
+		flicker_target = null
+	flicker_target = target
+	if(action == "Flicker")
+		to_chat(body,"<span class='warning'>You will attempt to flicker [target.body.real_name]'s room tonight.</span>")
+	else
+		to_chat(body,"<span class='danger'>You will hunt everyone in a flickering room down tonight.</span>")
+
+/datum/mafia_role/nightmare/proc/flicker_or_hunt(datum/mafia_controller/source)
+	if(game_status != MAFIA_ALIVE || !flicker_target)
+		return
+	if(SEND_SIGNAL(src,COMSIG_MAFIA_CAN_PERFORM_ACTION,source,"nightmare actions",flicker_target) & MAFIA_PREVENT_ACTION)
+		to_chat(flicker_target.body, "<span class='warning'>Your actions were prevented!</span>")
+		return
+	var/datum/mafia_role/R = flicker_target
+	flicker_target = null
+	if(R != src) //flicker
+		to_chat(R.body, "<span class='userdanger'>The lights begin to flicker and dim. You're in danger.</span>")
+		flickering += R
+	else //hunt
```

Comment:
Style thing, why not return here? I see you doing it elsewhere, and it'll yeet that else statement and it's indent

Reply:
yeah early return would probably do better. It didn't work when i was using flickering in this case but now I should really just do as you say